tuplex/utils/src/TypeSystem.cc

```diff
@@ -960,6 +960,103 @@ namespace python {
         return true;
     }
 
+    Type compatibleType(const python::Type &a, const python::Type &b, bool autoUpcast) {
```

Comment:
I think there's some function unifyTypes or so that does something quite similar...

Reply:
Merged into a single function unifyTypes in TypeSystem